### PR TITLE
[Enhancement] Add female and double grunts

### PR DESCRIPTION
**Pull request by Tempo-anon**
## What are the changes?
* Allowed female grunts to show up (cosmetic change)
* Allows evil team grunts to spawn as double battles
* Did some code refactoring
* Added localization for languages without the grunt translations

## Why am I doing these changes?
We had the assets for this but never implemented it

## What did change?
* `battle.ts`
  * Added documentation to `getRandomTrainerFunc`
  * Added a new optional `randomGender` variable to `getRandomTrainerFunc`
  * The previous double battle code was never working (and also had a bug) but I repurposed it to only work for evil team grunts
* Localizations
- [x] EN
- [x] DE
- [ ] ES
- [x] FR
- [x] IT
- [x] KO
- [ ] PT_BR
- [x] ZH_CN
- [x] ZH_TW

### Screenshots/Videos
Female grunts generating alongside male grunts

Double grunts

## How to test the changes?
Override to wave 61 

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [ ] Have I tested the changes (manually)?
    - [ ] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by Adri1 — APPROVED** [review]

**Review by frutescens — APPROVED** [review]
Think it looks good!

File: src/locales/it/trainers.ts [hunk 1]
```diff
@@ -118,7 +118,25 @@ export const trainerClasses: SimpleTranslationEntries = {
...
+  "aqua_grunts": "Aqua Grunts",
+  "galactic_grunt": "Galactic Grunt",
+  "galactic_grunt_female": "Galactic Grunt",
+  "galactic_grunts": "Galactic Grunts",
+  "plasma_grunt": "Plasma Grunt",
+  "plasma_grunt_female": "Plasma Grunt",
+  "plasma_grunts": "Plasma Grunts",
+  "flare_grunt": "Flare Grunt",
+  "flare_grunt_female": "Flare Grunt",
+  "flare_grunts": "Flare Grunts",
```
Comment (NicusPulcis) [inline src/locales/it/trainers.ts:139]:
```suggestion
  "rocket_grunt": "Recluta Team Rocket",
  "rocket_grunts": "Reclute Team Rocket",
  "rocket_grunt_female": "Recluta Team Rocket",
  "magma_grunt": "Recluta Team Magma",
  "magma_grunt_female": "Recluta Team Magma",
  "magma_grunts": "Reclute Team Magma",
  "aqua_grunt": "Recluta Team Idro",
  "aqua_grunt_female": "Recluta Team Idro",
  "aqua_grunts": "Recluta Team Idro",
  "galactic_grunt": "Recluta Team Galassia",
  "galactic_grunt_female": "Recluta Team Galassia",
  "galactic_grunts": "Reclute Team Galassia",
  "plasma_grunt": "Seguace Plasma",
  "plasma_grunt_female": "Seguace Plasma",
  "plasma_grunts": "Seguaci Plasma",
  "flare_grunt": "Recluta Team Flare",
  "flare_grunt_female": "Recluta Team Flare",
  "flare_grunts": "Reclute Team Flare",
```

**Review by NicusPulcis — APPROVED** [review]
Italian localization

File: src/locales/ko/trainers.ts [hunk 2]
```diff
@@ -127,16 +127,22 @@ export const trainerClasses: SimpleTranslationEntries = {
...
+  "aqua_grunts": "Aqua Grunts",
   "galactic_grunt": "갤럭시단 조무래기",
   "galactic_grunt_female": "갤럭시단 조무래기",
+  "galactic_grunts": "Galactic Grunts",
   "plasma_grunt": "플라스마단 조무래기",
   "plasma_grunt_female": "플라스마단 조무래기",
+  "plasma_grunts": "Plasma Grunts",
   "flare_grunt": "플레어단 조무래기",
   "flare_grunt_female": "플레어단 조무래기",
+  "flare_grunts": "Flare Grunts",
```
Comment (EnochG1) [inline src/locales/ko/trainers.ts:145]:
```suggestion
  "rocket_grunts": "로켓단 조무래기들",
  "magma_grunt": "마그마단 조무래기",
  "magma_grunt_female": "마그마단 조무래기",
  "magma_grunts": "마그마단 조무래기들",
  "aqua_grunt": "아쿠아단 조무래기",
  "aqua_grunt_female": "아쿠아단 조무래기",
  "aqua_grunts": "아쿠아단 조무래기들",
  "galactic_grunt": "갤럭시단 조무래기",
  "galactic_grunt_female": "갤럭시단 조무래기",
  "galactic_grunts": "갤럭시단 조무래기들",
  "plasma_grunt": "플라스마단 조무래기",
  "plasma_grunt_female": "플라스마단 조무래기",
  "plasma_grunts": "플라스마단 조무래기들",
  "flare_grunt": "플레어단 조무래기",
  "flare_grunt_female": "플레어단 조무래기",
  "flare_grunts": "플레어단 조무래기들",
```

File: src/locales/pt_BR/trainers.ts [hunk 3]
```diff
@@ -127,16 +127,22 @@ export const trainerClasses: SimpleTranslationEntries = {
...
+  "aqua_grunts": "Aqua Grunts",
   "galactic_grunt": "Recruta da Equipe Galáctica",
   "galactic_grunt_female": "Recruta da Equipe Galáctica",
+  "galactic_grunts": "Galactic Grunts",
   "plasma_grunt": "Recruta da Equipe Plasma",
   "plasma_grunt_female": "Recruta da Equipe Plasma",
+  "plasma_grunts": "Plasma Grunts",
   "flare_grunt": "Recruta da Equipe Flare",
   "flare_grunt_female": "Recruta da Equipe Flare",
+  "flare_grunts": "Flare Grunts",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/trainers.ts:145]:
```suggestion
  "rocket_grunts": "Recrutas da Equipe Rocket",
  "magma_grunt": "Recruta da Equipe Magma",
  "magma_grunt_female": "Recruta da Equipe Magma",
  "magma_grunts": "Recrutas da Equipe Magma",
  "aqua_grunt": "Recruta da Equipe Aqua",
  "aqua_grunt_female": "Recruta da Equipe Aqua",
  "aqua_grunts": "Recrutas da Equipe Aqua",
  "galactic_grunt": "Recruta da Equipe Galáctica",
  "galactic_grunt_female": "Recruta da Equipe Galáctica",
  "galactic_grunts": "Recrutas da Equipe Galáctica",
  "plasma_grunt": "Recruta da Equipe Plasma",
  "plasma_grunt_female": "Recruta da Equipe Plasma",
  "plasma_grunts": "Recrutas da Equipe Plasma",
  "flare_grunt": "Recruta da Equipe Flare",
  "flare_grunt_female": "Recruta da Equipe Flare",
  "flare_grunts": "Recrutas da Equipe Flare",
```

**Pull request merged**